server/src/lsp-server.spec.ts

```diff
@@ -845,3 +845,38 @@ export function factory() {
         );
     }).timeout(10000);
 });
+
+describe('diagnostics (no client support)', () => {
+    before(async () => {
+        // Remove the "textDocument.publishDiagnostics" client capability.
+        const clientCapabilitesOverride = getDefaultClientCapabilities();
+        delete clientCapabilitesOverride.textDocument?.publishDiagnostics;
+        server = await createServer({
```

Comment:
I'm not super familiar with all this, but is there a way to close these servers? As is, I think all the ones created will still be active until the entire test suite's process ends, and will contend for the same log files.

Not a blocking comment.

Reply:
The `beforeEach` hook closes the server: https://github.com/theia-ide/typescript-language-server/blob/5b3c3d9110140ab11617d3c5586498d8a18f7215/server/src/lsp-server.spec.ts#L28-L31

Those nothing seems to be closing the server after all tests finish.